### PR TITLE
Unify `deploy_local.sh` and `prepare_env.sh` as `install_deps.sh` to install dependencies of dev or k8s

### PR DESCRIPTION
**Pull request by acezen**
Signed-off-by: acezen <weibin.zen@gmail.com>

## What do these changes do?
- Unify `deploy_local.sh` and `prepare_env.sh` as `install_deps.sh` to install dependencies of dev or k8s.
- Build and deploy GraphScope with `make` command.
- Adapt wheel packaging to latest install_deps.sh

## Related issue number

related issue #709

### CODE REVIEW

File: .github/workflows/build-graphscope-wheels-linux.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ on:
       - '**.rst'
       - 'docs/**'
       - 'scripts/**'
-      - '!scripts/deploy_local.sh'
+      - '!scripts/install_deps.sh'
```
Comment (yecol) [inline .github/workflows/build-graphscope-wheels-linux.yml:14]: let's remove this line in all ci configurations before its merge.

File: .github/workflows/local.yml [hunk 2]
```diff
@@ -55,8 +55,21 @@ jobs:
       uses: mxschmitt/action-tmate@v2
 
     - name: Build GraphScope
+      env:
+        OS : ${{ matrix.os }}
```
Comment (yecol) [inline .github/workflows/local.yml:59]: Why do we still have the stage for building in this script?
The target of this renaming issue is to make the building process clear and easy:
A user uses this script to INSTALL dependencies ONLY.

Then, the one can only use a single command 
`make graphscope` to build the artifacts by himself.

**Review by yecol — CHANGES_REQUESTED** [review]

File: scripts/install_deps.sh [hunk 3]
```diff
@@ -17,16 +17,14 @@ readonly GRAPE_BRANCH="master" # libgrape-lite branch
 readonly V6D_VERSION="0.3.4"  # vineyard version
 readonly V6D_BRANCH="v0.3.4" # vineyard branch
 
-readonly SOURCE_DIR="$( cd "$(dirname $0)/.." >/dev/null 2>&1 ; pwd -P )"
 readonly OUTPUT_ENV_FILE="${HOME}/.graphscope_env"
 IS_IN_WSL=false && [[ ! -z "${IS_WSL}" || ! -z "${WSL_DISTRO_NAME}" ]] && IS_IN_WSL=true
 readonly IS_IN_WSL
-INSTALL_PREFIX=/opt/graphscope
+INSTALL_PREFIX="/usr/local"
```
Comment (yecol) [inline scripts/install_deps.sh:23]: DEPS_PREFIX

File: scripts/install_deps.sh [hunk 4]
```diff
@@ -883,101 +964,78 @@ install_deps() {
...
-  succ "GraphScope has been built successfully and installed on ${INSTALL_PREFIX}."
-  succ "Please manually run:"
-  succ "export GRAPHSCOPE_HOME=${INSTALL_PREFIX}"
-  succ "before using GraphScope via Python client, enjoy!"
-  if [[ ${VERBOSE} == true ]]; then
-    set +x
-  fi
+  pull_images
+
+  succ_msg="The script has successfully prepared an environment for GraphScope.
```
Comment (yecol) [inline scripts/install_deps.sh:1018]: succ_msg_dev:
"The script has installed all dependencies for building GraphScope, use command \
`$ export XXX`
`$ make graphscope`
to build and develop graphscope."

succ_msg_k8s=
"The script has prepared a local k8s cluster with **kind** to run GraphScope in distributed mode, export essential environment variables with \
`$ export xxxxxx`
before using GraphScope. "

File: .github/workflows/local.yml [hunk 5]
```diff
@@ -55,8 +55,21 @@ jobs:
...
       run: |
-        /bin/bash ${GITHUB_WORKSPACE}/scripts/deploy_local.sh build_and_deploy --verbose --build_type debug
+        pushd ${GITHUB_WORKSPACE}
+        source ${HOME}/.graphscope_env
+        if [ "${OS}" == "macos-10.15" ];
+        then
+            # need libomp.dylib to find MPI_CXX
+            export LDFLAGS=-L/usr/local/opt/llvm/lib
```
Comment (yecol) [inline .github/workflows/local.yml:66]: why not in L-62?

File: .github/workflows/local.yml [hunk 6]
```diff
@@ -55,8 +55,21 @@ jobs:
...
+            make graphscope WITH_LEARNING_ENGINE=OFF NETWORKX=OFF BUILD_TYPE=debug
+        else
+            export WITH_LEARNING_ENGINE=ON
```
Comment (yecol) [inline .github/workflows/local.yml:69]: why this? what's the default value?

File: .github/workflows/local.yml [hunk 7]
```diff
@@ -55,8 +55,21 @@ jobs:
...
+            make graphscope NETWORKX=OFF BUILD_TYPE=debug
```
Comment (yecol) [inline .github/workflows/local.yml:70]: why networkx OFF? is that the default value?

File: .github/workflows/local.yml (hunk 6 above)
Comment (acezen) [reply .github/workflows/local.yml:69]: The default value is `ON`, but if graphscope-client packaging with graph-learn, it  need the environment `WITH_LEARNING_ENGINE` to be set `ON` .  https://github.com/alibaba/GraphScope/blob/a32e711d6dc2fd8360e5f6817fdfd38a64813719/python/setup.py#L134

File: .github/workflows/local.yml (hunk 7 above)
Comment (acezen) [reply .github/workflows/local.yml:70]: we don't test networkx in local ci. so we set networkx OFF to save the time of bulding GraphScope.

**Comment by yecol** [comment]
1. add a --cn option, export brew sources for speed;
2. for jdk, maybe use mirror of aliyun OSS for speed. (hetao is uploading one for macOS)

File: scripts/install_deps.sh [hunk 8]
```diff
@@ -649,21 +698,27 @@ install_dependencies() {
...
 
   elif [[ "${PLATFORM}" == *"Darwin"* ]]; then
+    if [[ ${CN_MIRROR} == true ]]; then
+      # set brew to tsinghua mirror
+      export HOMEBREW_BREW_GIT_REMOTE="https://mirrors.tuna.tsinghua.edu.cn/git/homebrew/brew.git"
+      export HOMEBREW_CORE_GIT_REMOTE="https://mirrors.tuna.tsinghua.edu.cn/git/homebrew/homebrew-core.git"
+      export HOMEBREW_BOTTLE_DOMAIN="https://mirrors.tuna.tsinghua.edu.cn/homebrew-bottles"
+    fi
     log "Installing packages ${BASIC_PACKGES_TO_INSTALL[*]}"
     brew install ${BASIC_PACKGES_TO_INSTALL[*]}
```
Comment (yecol) [inline scripts/install_deps.sh:708]: Did you test the brew? I think it needs `brew update` to make it works?

File: scripts/install_deps.sh [hunk 9]
```diff
@@ -649,21 +698,27 @@ install_dependencies() {
...
 
-    if [[ "${packages_to_install[@]}" =~ "jdk" ]]; then
-      # packages_to_install contains jdk8
-      log "Installing adoptopenjdk11."
-      wget -c https://github.com/adoptium/temurin11-binaries/releases/download/jdk-11.0.12%2B7/OpenJDK11U-jdk_x64_mac_hotspot_11.0.12_7.pkg \
+    if [[ ${CN_MIRROR} == true && "${packages_to_install[*]}" =~ "openjdk@11" ]]; then
```
Comment (yecol) [inline scripts/install_deps.sh:710]: how about CN_MIRROR false?

**Review by yecol — CHANGES_REQUESTED** [review]

**Review by yecol — APPROVED** [review]

**Pull request merged**